src/client/main.js

```diff
@@ -119,7 +118,8 @@ function displayBlock(root, value) {
 }
 
 export function undefine(id) {
-  cellsById.get(id)?.variables.forEach((v) => v.delete());
+  clear(rootsById.get(id));
+  cellsById.get(id).variables.forEach((v) => v.delete());
```

Comment:
I removed the optional chaining here because we should only be undefining cells that were defined (and we want to notice if our logic is broken).